tsl/src/continuous_aggs/README.md

```diff
@@ -115,19 +125,70 @@ entries that are outside the refresh window. Subsequently, if the
+aggregate.
+
+To avoid dependencies on the TimescaleDB extension, the invalidation
+plugin is made as simple as possible and does not use the metadata
+tables of the TimescaleDB extension and instead is configured to read
+specific relids and attribute names when fetching data.
+
+While processing the WAL, the timestamps are collected into ranges and
+(similarly to when using the trigger to record changes) only one
+invalidation range (lowest and highest modified value in microseconds
+since UNIX Epoch) is produced for each modified hypertable in each
+mutating transaction. For the same reasons as above, this means that a
+refresh might materialize more data than necessary.
+
+The modifications are represented as tuples consisting of the relid
+for the change and the start and end timestamp as int64, which is the
+internal representation of timestamp values. It is up to the receiver
+to use the relid and timestamps correctly.
+
+To be able to collect a unique key for each changed row when doing a
+`DELETE` or an `UPDATE`, it is required that the hypertable has a
+replica identity set. This can either be the primary key of the
+hypertable, or a unique index, or the full row if neither is suitable.
+
 ## Distribution of functions across files
-common.c
-This file contains the functions common in all scenarios of creating a continuous aggregates.
 
-create.c
-This file contains the functions that are directly responsible for the creation of the continuous aggregates,
-like creating hypertable, catalog_entry, view, etc.
+Each source file has an associated header file that should be included
+to use functions from the corresponding file.
+
+<dl>
+<dt>`common.c`</dt>
+<dd>This file contains the functions common in all scenarios of creating a continuous aggregates.</dd>
+
+<dt>`create.c`</dt>
+<dd>This file contains the functions that are directly responsible for the creation of the continuous aggregates,
+like creating hypertable, catalog_entry, view, etc.</dd>
+
+<dt>`finalize.c`</dt>
+<dd>This file contains the specific functions for the case when continous aggregates are created in old format.</dd>
+
+<dt>`materialize.c`</dt>
+<dd>This file contains the functions directly dealing with the materialization of the continuous aggregates.</dd>
+
+<dt>`repair.c`</dt>
+<dd>The repair and rebuilding related functions are put together in this file.</dd>
+</dl>
+
+<dt>`invalidation.c`</dt>
+<dd>Functions related to invalidation processing for continuous aggregates.</dd>
+</dl>
+
+<dt>`invalidation_plugin.c`</dt>
+<dd>The invalidation plugin.</dd>
 
-finalize.c
-This file contains the specific functions for the case when continous aggregates are created in old format.
+<dt>`invalidation_cache.c`</dt>
+<dd>The invalidation cache for the invalidation plugin.</dd>
 
-materialize.c
-This file contains the functions directly dealing with the materialization of the continuous aggregates.
+<dt>`invalidation_funcs.c`</dt>
```

Comment:
This file doesn't exist anymore.

Reply:
It changed name to `invalidation_record.c` since that's all that remained after.